be/src/olap/merger.cpp

```diff
@@ -21,107 +21,55 @@
 #include <vector>
 
 #include "olap/olap_define.h"
-#include "olap/rowset/segment_group.h"
 #include "olap/tablet.h"
 #include "olap/reader.h"
 #include "olap/row_cursor.h"
 
-using std::list;
-using std::string;
-using std::unique_ptr;
-using std::vector;
-
 namespace doris {
 
-Merger::Merger(TabletSharedPtr tablet, RowsetWriterSharedPtr writer, ReaderType type) :
-        _tablet(tablet),
-        _output_rs_writer(writer),
-        _reader_type(type),
-        _row_count(0) {}
-
-Merger::Merger(TabletSharedPtr tablet, ReaderType type, RowsetWriterSharedPtr writer,
-               const std::vector<RowsetReaderSharedPtr>& rs_readers) :
-        _tablet(tablet),
-        _output_rs_writer(writer),
-        _input_rs_readers(rs_readers),
-        _reader_type(type),
-        _row_count(0) {}
-
-OLAPStatus Merger::merge(const vector<RowsetReaderSharedPtr>& rs_readers,
-                         int64_t* merged_rows, int64_t* filted_rows) {
-    _input_rs_readers = rs_readers;
-    OLAPStatus res = merge();
-    if (res == OLAP_SUCCESS) {
-        *merged_rows= _merged_rows;
-        *filted_rows = _filted_rows;
-    }
-    return res;
-}
-
-OLAPStatus Merger::merge() {
-    // Create and initiate reader for scanning and multi-merging specified
-    // OLAPDatas.
+OLAPStatus Merger::merge_rowsets(TabletSharedPtr tablet,
+                                 ReaderType reader_type,
+                                 const std::vector<RowsetReaderSharedPtr>& src_rowset_readers,
+                                 RowsetWriter* dst_rowset_writer,
+                                 Merger::Statistics* stats_output) {
     Reader reader;
     ReaderParams reader_params;
-    reader_params.tablet = _tablet;
-    reader_params.reader_type = _reader_type;
-    reader_params.rs_readers = _input_rs_readers;
-    reader_params.version = _output_rs_writer->version();
-
-    if (OLAP_SUCCESS != reader.init(reader_params)) {
-        LOG(WARNING) << "fail to initiate reader. tablet=" << _tablet->full_name();
-        return OLAP_ERR_INIT_FAILED;
-    }
+    reader_params.tablet = std::move(tablet);
```

Comment:
Better not to move tablet, because tablet may be used in the following code.

Reply:
ok